nemo/collections/nlp/models/duplex_text_normalization/duplex_tagger.py

```diff
@@ -0,0 +1,315 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import time
+import nltk
+import nemo.collections.nlp.data.text_normalization.constants as constants
+nltk.download('punkt')
+
+from typing import List, Optional
+from pytorch_lightning import Trainer
+from omegaconf import DictConfig
+
+from torch import nn
+from transformers import AutoModelForTokenClassification, AutoTokenizer, DataCollatorForTokenClassification
+
+from nemo.utils import logging
+from nemo.collections.nlp.models.nlp_model import NLPModel
+from nemo.core.classes.common import PretrainedModelInfo
+
+from nemo.collections.nlp.models.duplex_text_normalization.utils import has_numbers
+from nemo.collections.nlp.data.text_normalization import TextNormalizationTaggerDataset
+
+__all__ = ['DuplexTaggerModel']
+
+class DuplexTaggerModel(NLPModel):
+    """
+    Transformer-based (duplex) tagger model for TN/ITN.
+    """
+
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        self._tokenizer = AutoTokenizer.from_pretrained(cfg.tokenizer, add_prefix_space=True)
+        super().__init__(cfg=cfg, trainer=trainer)
+        self.num_labels = len(constants.ALL_TAG_LABELS)
+        self.model = AutoModelForTokenClassification.from_pretrained(cfg.transformer,
```

Comment:
could you use nemo tokenclassifcation?

Reply:
So first of all, I cannot directly use the TokenClassificationModel out of the box because the model expects a different data format from the typical format of a TN dataset.

So there are two other ways: (1) make DuplexTaggerModel become a child class of TokenClassificationModel  (2) initialize an instance of TokenClassificationModel  inside DuplexTaggerModel.

Either way, I will still need to write the same amount of additional code like this commit. This is because my model has some other different logics (e.g, adding prefix depending on tasks, some postprocessing of the predicted tags, different DataLoader, ...). 

Given that TokenClassificationModel is also just a wrapper around HuggingFace's AutoModelForTokenClassification, I don't think it is worth using it at this point.